src/azure-cli-core/azure/cli/core/commands/__init__.py

```diff
@@ -215,12 +228,18 @@ def register_extra_cli_argument(command, dest, **kwargs):
     '''
     _cli_extra_argument_registry[command][dest] = CliCommandArgument(dest, **kwargs)
 
+def cli_command_with_handler(module_name, name, operation, client_factory=None, transform=None, table_transformer=None):
```

Comment:
Caching has downsides:
- cache can become stale
- how do we keep the cache correct
- when do we refresh it
- where do we save the cache

I didn't want to use caching unless we *really have to* (e.g. if tab completion is still considered too slow after this change).

For users that don't use tab completion, the cache would have a minimal benefit as a network request can take 0.5 seconds anyway.